include/mockturtle/io/write_blif.hpp

```diff
@@ -85,95 +90,139 @@ void write_blif( Ntk const& ntk, std::ostream& os )
   static_assert( has_node_to_index_v<Ntk>, "Ntk does not implement the node_to_index method" );
   static_assert( has_node_function_v<Ntk>, "Ntk does not implement the node_function method" );
 
-  topo_view topo_ntk{ntk};
+  // topo_view topo_ntk{ntk};
 
   /* write model */
-  os << ".model netlist\n";
+  os << ".model top\n";
 
   /* write inputs */
   if ( topo_ntk.num_pis() > 0u )
   {
       os << ".inputs ";
-      topo_ntk.foreach_pi( [&]( auto const& n ) {
+      topo_ntk.foreach_ci( [&]( auto const& n, auto index ) {
+        if ( ( ( index + 1 ) <= topo_ntk.num_cis() - topo_ntk.num_latches() ) ) {
           if constexpr ( has_has_name_v<Ntk> && has_get_name_v<Ntk> )
           {
             signal<Ntk> const s = topo_ntk.make_signal( topo_ntk.node_to_index( n ) );
-            std::string const name = topo_ntk.has_name( s ) ? topo_ntk.get_name( s ) : fmt::format( "c_{}", topo_ntk.get_node( s ) );
+            std::string const name = topo_ntk.has_name( s ) ? topo_ntk.get_name( s ) : fmt::format( "pi{}", topo_ntk.get_node( s ) );
             os << name << ' ';
           }
           else
           {
-            os << fmt::format( "c_n{} ", topo_ntk.node_to_index( n ) );
+            os << fmt::format( "pi{} ", topo_ntk.node_to_index( n ) );
```

Comment:
Do you mean "new_n{}" instead of "pi{}"?

Reply:
Same reason as the comment above.